integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/predicate/AndPredicateSpecificsIT.java

```diff
@@ -0,0 +1,270 @@
+import org.hibernate.search.engine.search.predicate.SearchPredicate;
+import org.hibernate.search.integrationtest.backend.tck.testsupport.util.rule.SearchSetupHelper;
+import org.hibernate.search.util.impl.integrationtest.mapper.stub.BulkIndexer;
+import org.hibernate.search.util.impl.integrationtest.mapper.stub.SimpleMappedIndex;
+import org.hibernate.search.util.impl.integrationtest.mapper.stub.StubMappingScope;
+
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+public class AndPredicateSpecificsIT {
+
+	private static final String DOCUMENT_1 = "1";
+	private static final String DOCUMENT_2 = "2";
+	private static final String DOCUMENT_3 = "3";
+
+	// Document 1
+
+	private static final String FIELD1_VALUE1 = "Irving";
+	private static final Integer FIELD2_VALUE1 = 3;
+	private static final Integer FIELD3_VALUE1 = 4;
+	private static final Integer FIELD4_VALUE1AND2 = 1_000;
+	private static final Integer FIELD5_VALUE1AND2 = 2_000;
+
+	// Document 2
+
+	private static final String FIELD1_VALUE2 = "Auster";
+	private static final Integer FIELD2_VALUE2 = 13;
+	private static final Integer FIELD3_VALUE2 = 14;
+	// Field 4: Same as document 1
+	// Field 5: Same as document 1
+
+	// Document 3
+
+	private static final String FIELD1_VALUE3 = "Coe";
+	private static final Integer FIELD2_VALUE3 = 25;
+	private static final Integer FIELD3_VALUE3 = 42;
+	private static final Integer FIELD4_VALUE3 = 42_000; // Different from document 1
+	private static final Integer FIELD5_VALUE3 = 142_000; // Different from document 1
+
+	@ClassRule
+	public static final SearchSetupHelper setupHelper = new SearchSetupHelper();
+
+	private static final SimpleMappedIndex<IndexBinding> index = SimpleMappedIndex.of( IndexBinding::new );
+
+	@BeforeClass
+	public static void setup() {
+		setupHelper.start().withIndex( index ).setup();
+
+		initData();
+	}
+
+	@Test
+	public void empty() {
+		assertThatQuery( index.query()
+				.where( f -> f.and() ) )
+				.hasNoHits();
+	}
+
+	@Test
+	public void and() {
+		assertThatQuery( index.query()
+				.where( f -> f.and( f.match().field( "field1" ).matching( FIELD1_VALUE1 ) ) ) )
+				.hasDocRefHitsAnyOrder( index.typeName(), DOCUMENT_1 );
+
+		assertThatQuery( index.query()
+				.where( f -> f.and(
+						f.match().field( "field1" ).matching( FIELD1_VALUE1 ),
+						f.match().field( "field2" ).matching( FIELD2_VALUE2 )
+				) ) )
+				.hasNoHits();
+
+//		// TODO YB: Should not compile
+//		assertThatQuery( index.query()
+//				.where( f -> f.and( f.match().field( "field1" ).matching( FIELD1_VALUE1 ) )
+//						.add( f.match().field( "field2" ).matching( FIELD2_VALUE2 ) ) ) )
+//				.hasNoHits();
+//
+//		// TODO YB: Should not compile
+//		assertThatQuery( index.query()
+//				.where( f -> f.and( f.match().field( "field1" ).matching( FIELD1_VALUE1 ) )
+//						.with( b -> b.add( f.match().field( "field2" ).matching( FIELD2_VALUE2 ) ) ) ) )
+//				.hasNoHits();
```

Comment:
Unless you intend to do something about it, you should probably remove this commented-out code.

Reply:
Oops!  I never intended to commit that!
I have stashed my secret... stash of code that should not compile!  ;-)